.github/workflows/js-checks.yml

```
@@ -19,6 +19,8 @@ jobs:
       contents: read
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

What if we let the caller decide the fetch depth? That can be done by simply adding an input `fetch-depth` that defaults to `1` (number). Then change this line as follows:

```suggestion
          fetch-depth: $${{ inputs.fetch-depth }}
```

---

Applied here [e4511f3](https://github.com/hemilabs/actions/pull/12/commits/e4511f30c243e44d21c812dc18f777dc1e7eefb0)